### PR TITLE
Feature/uniswap and conditions

### PR DESCRIPTION
**Pull request by zeroknots**
Adding uniswap and time based `ICondition` scheduling to moduleKit

### CODE REVIEW

File: src/modulekit/conditions/ScheduleCondition.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { ICondition } from "../IExecutor.sol";
+import "./helpers/ChainlinkTokenPrice.sol";
+
+import "forge-std/console2.sol";
```
Comment (kopy-kat) [inline src/modulekit/conditions/ScheduleCondition.sol:7]: Let’s remove this

File: src/modulekit/conditions/ScheduleCondition.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    function scheduleIsDue(
+        Schedule storage scheduleForAccount,
+        Params memory params
+    )
+        private
+        view
+        returns (bool isDue)
+    {
+        uint256 lastExecuted = scheduleForAccount.lastExecuted;
+        console2.log("last exec:", lastExecuted);
```
Comment (kopy-kat) [inline src/modulekit/conditions/ScheduleCondition.sol:48]: Let’s remove this too

File: src/modulekit/conditions/ScheduleCondition.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import "forge-std/console2.sol";
+
+contract ScheduleCondition is ICondition, ChainlinkTokenPrice {
+    struct Params {
+        uint256 triggerEveryHours;
+    }
+
+    struct Schedule {
+        uint48 lastExecuted;
```
[the remark this answers is not in the document]
Comment (zeroknots) [reply src/modulekit/conditions/ScheduleCondition.sol]: addressed in
https://github.com/rhinestonewtf/modulekit/pull/15/commits/ae01679f37ad307402854165355e943ae25b7095

File: src/modulekit/conditions/ScheduleCondition.sol (hunk 2 above)
Comment (zeroknots) [reply src/modulekit/conditions/ScheduleCondition.sol]: addressed in
https://github.com/rhinestonewtf/modulekit/pull/15/commits/ae01679f37ad307402854165355e943ae25b7095

File: src/modulekit/conditions/ScheduleCondition.sol (hunk 1 above)
Comment (zeroknots) [reply src/modulekit/conditions/ScheduleCondition.sol]: addressed in
https://github.com/rhinestonewtf/modulekit/pull/15/commits/ae01679f37ad307402854165355e943ae25b7095

File: src/modulekit/integrations/uniswap/v3/UniswapSwaps.sol [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+pragma solidity ^0.8.19;
+
+import "../helpers/MainnetAddresses.sol";
+import "../../interfaces/uniswap/v3/ISwapRouter.sol";
+import "../../ERC20Actions.sol";
+import "../../../IExecutor.sol";
+
+/// @author zeroknots
+
+library UniswapSwaps {
```
[the remark this answers is not in the document]
Comment (kopy-kat) [reply src/modulekit/integrations/uniswap/v3/UniswapSwaps.sol:11]: I see and calling it UniswapSwaps sounds bad lol

Ok lets leave it for now and see later

**Pull request merged**